cpp/src/parquet/column_reader.cc

```diff
@@ -1141,12 +1146,14 @@ int64_t TypedColumnReaderImpl<DType>::ReadBatchSpaced(
 template <typename DType>
 int64_t TypedColumnReaderImpl<DType>::Skip(int64_t num_values_to_skip) {
   int64_t values_to_skip = num_values_to_skip;
-  while (HasNext() && values_to_skip > 0) {
+  // Optimization: Do not call HasNext() when values_to_skip == 0.
+  while (values_to_skip > 0 && HasNext()) {
     // If the number of values to skip is more than the number of undecoded values, skip
     // the Page.
-    if (values_to_skip > (this->num_buffered_values_ - this->num_decoded_values_)) {
-      values_to_skip -= this->num_buffered_values_ - this->num_decoded_values_;
-      this->num_decoded_values_ = this->num_buffered_values_;
+    const int64_t available_values = this->available_values_current_page();
+    if (values_to_skip >= available_values) {
```

Comment:
This line has the main change for this request.

Reply:
I see, thanks.